File: java/howmanydigits.java
import java.util.Scanner;;

public class howmanydigits {
	static final double[] bases = new double[1000001];

	static {
		bases[0] = bases[1] = 1;
		for (int i = 2; i < bases.length; i++)
			bases[i] = Math.log10(i) + bases[i - 1];
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		while (scan.hasNextInt())
			System.out.println((int) Math.floor(bases[scan.nextInt()]));
		scan.close();
	}
}

File: ts/src/howmanydigits.ts
import readline = require('readline')

function generateBases (): number[] {
  const result: number[] = []
  const size = 1000001
  result.push(1)
  result.push(1)
  for (let i = 2; i < size; i++) {
    result.push(Math.log10(i) + (result[result.length - 1] ?? 0))
  }
  return result
}

const bases: number[] = generateBases()

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    const num = parseInt(line)
    console.log(Math.floor(bases[num] ?? 0))
  })
  .on('close', () => {
    process.exit(0)
  })
